CMakeLists.txt

```diff
@@ -1157,16 +1157,16 @@ endif (UNIX)
 #   Release => -O3 -NDEBUG
 # Right now we only support gcc and cl but could change in future
 if (DEBUG)
-  set(CMAKE_C_FLAGS "${BASE_CFLAGS} ${BASE_CONLY_FLAGS} ${DBG} ${DBG_OPT}")
-  set(CMAKE_CXX_FLAGS "${BASE_CFLAGS} ${BASE_CXXONLY_FLAGS} ${DBG} ${DBG_OPT}")
+  set(CMAKE_C_FLAGS "${BASE_CFLAGS} ${BASE_CONLY_FLAGS} ${DBG} ${DBG_OPT} $ENV{CFLAGS}")
```

Comment:
I think it's better to define a new environment variable for these additional options. `ADDITIONAL_CFLAGS` maybe. This flag can then be set before invoking CMake. Does that still solve your use-case?

Reply:
Yes, it would solve my use-case, but why leave CFLAGS broken and introduce a new variable that nobody knows about?

Reply:
I was just concerned about our build picking up any flags from the env var that we don't intend to. Using a new env var and also documenting it in `api/docs/building.dox` would be the safest I think.

Reply:
Derek's comment on the issue (https://github.com/DynamoRIO/dynamorio/issues/5266#issuecomment-1001689348) seems to agree with using a new environment variable. Can you make the change?

Reply:
I think using `CFLAGS` seems good since that's the method for passing flags like `-m32` today.  Since it is an explicit user-set list of flags, if the user includes flags that are not compatible with DR (or DrMemory) it is a user error, as opposed to the default sets of initial flags which we need to carefully handle.  We would assume the user knows what they are doing for `CFLAGS`.

Reply:
Oh, now I see how `CFLAGS=-m32` works today even that we don't add whole `CFLAGS` to the compilation flags.